### PR TITLE
Tic-Tac-Toe Milestone 3: Game Logic!

### PR DESCRIPTION
**Pull request by lmaldonadoch**
On this milestone, we added logic to our game. We Created 2 Classes (Player and GameLogic) and 1 Module (Board) which interact with each other to create the game flow and provide a good gaming experience for the user along with industry-standard quality code.
## Project milestones
- Create a specific branch with the name of game_logic.
- Add real logic for your game, get rid of all inline comments, and mocked logic used in the previous milestone.
-As main.rb file should be responsible for user interaction, this is the only(!) one place where puts/print and get.chomp can be used.
- Think about how you can encapsulate different parts of the game logic in classes. Do not create one class for everything.
- Read about DRY rule and use it.
- Think about the difference between attr_readers/attr_writers and attr_accessors. Use attr_accessors only if you really need it.
- Remember about the single responsibility rule. Always.
- Remember to cover edge cases: think what player could try to do to cheat the game and make sure that your program does not allow to perform invalid action.

screenshot3

screenshot1

screenshot2

## General requirements:
- [x] Add a descriptive README file to your repositories as described here.
- [x] Use meaningful names for your Pull Requests and add a short summary of its content as described here.
- [x] Use descriptive commit messages as described here.
- [x] Do not commit files that are not related to the project. Instead, add them to .gitignore file.
- [x] Do not commit inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.
## Git Flow:
- [x] Use a specific branch for each feature.
- [x] Use a descriptive name for your branches.
- [x] Use Rubocop to validate your code.

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Good Job team
Your code looks good and the game works fine. However, we have a minor change to make before we can merge. 
- [ ] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). Remove the `puts` statements on lines `15-17` of the `Board` module.
`puts` should only live in the `bin/main` file so move them there.


### Martin Afani

**Comment by lmaldonadoch** [comment]
Hi! @whiz25 Thank you very much for your comments. Here are the solutions:

- [x] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). Remove the `puts` statements on lines `15-17` of the `Board` module.
`puts` should only live in the `bin/main` file so move them there.
> Done! Thank you for the comment, you are very right. We should not have had any prints outside of main, so what we did is push the string into an array and return it. That way we were able to print it on main.

Thank you again and happy coding! :)

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module Board
+  def draw(player1_pos, player2_pos)
```
Comment (uryelah) [inline lib/board.rb:2]: - [ ] The #draw method from the module Board seems to be doing more than one thing:
1. fill the board
  a. check if board cell is filled by player 1 or 2
2. draw the board at the second each loop

You can make this at least two different methods, and ideally, also make the checking if included in player_pos another separated method too.

That would make your code more solid, be truer to the single responsibility rule, and easier to test in the future.

File: lib/gamelogic.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @game_finished = false
+    @continue_playing = true
+  end
+
+  def win
+    return @player1.name if @player1.check_winner
+    return @player2.name if @player2.check_winner
```
Comment (uryelah) [inline lib/gamelogic.rb:17]: - [ ] In this method and some other ones from this class you repeat very similar lines pertaining to the players, maybe you could think of a way to refactor how you store the players so you don't have to repeat the code so much inside the method. 
Think about how to be drier.

**Review by uryelah — APPROVED** [review]
Hello **Fernando Bahamondes Correa & Luis Maldonado Chávez**, I'm Sarah, your last reviewer for this PR 👩‍🚀  

Great job with this project and making the required fix from the last review!

Your project is approved but I added a couple of optional suggestions in the comments below if you want to refactor a bit (it could make your life a lot easier when you have to test your code).

---------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being awesome devs and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**